### PR TITLE
LibJS: Even more normative Temporal spec changes

### PR DESCRIPTION
**Pull request by linusg**
There's more, and I don't want to make tons of small PRs, so I'll draft this for now and accumulate more changes.

### CODE REVIEW

**Comment by linusg** [comment]
Well, I was hoping to get https://github.com/tc39/proposal-temporal/pull/1799 in here as well but apparently not anytime soon, so let's just get this merged.

File: Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp [hunk 1]
```diff
@@ -546,12 +547,64 @@ JS_DEFINE_NATIVE_FUNCTION(CalendarPrototype::fields)
...
+        auto next = iterator_step(global_object, *iterator_record);
+        if (vm.exception())
+            return {};
+
+        // b. If next is not false, then
+        if (!next)
+            break;
+
+        // i. Let nextValue be ? IteratorValue(next).
+        auto next_value = iterator_value(global_object, *next);
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp:573]: missing exception check

File: Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp [hunk 2]
```diff
@@ -219,6 +219,8 @@ JS_DEFINE_NATIVE_FUNCTION(CalendarPrototype::date_add)
     if (vm.exception())
         return {};
 
+    // TODO: Use BalanceDuration before AddISODate, see https://github.com/tc39/proposal-temporal/commit/6178ed35400fb56ef4013ce9d0163212ec096e57#diff-1e3b606bd8b3ef5c33eaf7b43afdfc240b0a926c8d0a66e02bfc9e1a7e8de8a2R798-R799
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp:222]: `BalanceDuration` was added in my latest PR :)

File: Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp [hunk 3]
```diff
@@ -270,6 +270,16 @@ Duration* create_temporal_duration(GlobalObject& global_object, double years, do
     return object;
 }
 
+// 7.5.8 CreateNegatedTemporalDuration ( duration ), https://tc39.es/proposal-temporal/#sec-temporal-createnegatedtemporalduration
+Duration* create_negated_temporal_duration(GlobalObject& global_object, Duration& duration)
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp:274]: `Duration const&`

**Review by IdanHo — COMMENTED** [review]
Just a couple of small issues:

File: Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp:577]: fixed!

File: Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/CalendarPrototype.cpp]: hah I didn't feel like implementing that at the time, nice :D

**Review by IdanHo — APPROVED** [review]
LGTM

**Pull request merged**